File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot implements Parrot {
    protected final int numberOfCoconuts;

    public AfricanParrot(int numberOfCoconuts) {
        this.numberOfCoconuts = numberOfCoconuts;
    }

    private double getLoadFactor() {
        return 9.0;
    }

    @Override
    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - getLoadFactor() * numberOfCoconuts);
    }

    @Override
    public String getCry() {
        return "Sqaark!";
    }
}

File: Kotlin/src/main/kotlin/parrot/AfricanParrot.kt
package parrot

import kotlin.math.max

class AfricanParrot(private val numberOfCoconuts: Int) : Parrot {
    override val speed: Double
        get() = max(0.0, baseSpeed - loadFactor * numberOfCoconuts)

    override val cry: String
        get() = "Sqaark!"

    private val loadFactor: Double
        get() = 9.0

}